src/torrent_info.cpp

```diff
@@ -374,9 +374,13 @@ namespace {
 		// symlinks have an implied "size" of zero. i.e. they use up 0 bytes of
 		// the torrent payload space
 		std::int64_t const file_size = (file_flags & file_storage::flag_symlink)
-			? 0
-			: dict.dict_find_int_value("length", -1);
-		if (file_size < 0)
+			? 0 : dict.dict_find_int_value("length", -1);
+
+		// if a file is too big, it will cause integer overflow in our
+		// calculations of the size of the merkle tree (which is all 'int'
+		// indices)
+		if (file_size < 0
+			|| (file_size / default_block_size) >= std::numeric_limits<int>::max())
```

Comment:
this is the actual fix

Reply:
The limit should be half this because `int` is used for the node index and the size of the merkle tree is 2n-1 nodes.

Reply:
ah, and removing the piece layer from the test torrent prevented that code from being exercised. I fixed this and added the piece layer back